### ISSUE TITLE
Setup newrelic instrumentation with from_definition

### ISSUE THREAD

**Issue by akshah123** [rmosolgo/graphql-ruby#1415]
Right now, we are using something like following to setup schema

```ruby
Appschema= GraphQL::Schema.from_definition(
  schema,
  default_resolve: RootResolver
)
```

How do I enable newrelic to work with above setup?
```ruby
Appschema = GraphQL::Schema.from_definition(
  schema,
  default_resolve: RootResolver
)

Appschema.define do
  use(GraphQL::Tracing::NewRelicTracing)
end
```

I did try following, but that doesnt provide details as listed here: http://graphql-ruby.org/queries/tracing.html#new-relic

**Comment by sync** [comment]
```
AppSchema.redefine {
    use(GraphQL::Tracing::NewRelicTracing)
  }
```

**Comment by akshah123** [comment]
Following does work.  I expected it to change the web transaction name but this simply adds breakdown of resolvers within GraphQL controller's time.

```ruby
Appschema.define do
  use(GraphQL::Tracing::NewRelicTracing)
end
```

It would be great if web transaction itself showed as the actual query or mutation.

**Comment by rmosolgo** [comment]
:+1: For either modifying the schema in-place, using `.define`: 

```ruby 
Appschema.define do
  use(GraphQL::Tracing::NewRelicTracing)
end
```

Or making a copy of it, where the copy of it has the tracing applied, using `.redefine`: 

```ruby
Appschema = GraphQL::Schema.from_definition(schema, default_resolve: RootResolver).redefine { 
  use(GraphQL::Tracing::NewRelicTracing)
}
```

Regarding the transaction name, sorry about that! It looks like a screenshot from the _old_ monitoring plug-in, when it was part of graphql-pro. 

Back then, someone reported an interesting issue: they had some pages that ran _multiple_ GraphQL queries, but since the plugin was resetting the transaction name, those queries _weren't_ grouped in the UI, so they couldn't tell how long the whole page took to render. 

But, it sounds like you would like each query to get its own transaction name, is that right? If so, I bet we can extend the plugin to support that option.

**Comment by akshah123** [comment]
@rmosolgo totally see the point about multiple queries/mutation being made in a single api call.

I do wonder if it would be possible to only change transaction name when it was a single query/mutation and if there are multiple keeping it graphqlcontroller/multiple or something similar.

**Comment by rmosolgo** [comment]
I'm not sure how to _detect_ whether a given query will be followed by other queries or not. Can you think of a way? 

Another straightforward option would be to switch on a flag in `context`. For example, if `context[:new_relic_transaction]` is true, then we set the transaction name. That way, in your controller, you could add 

```ruby 
context = {
  new_relic_transaction: true, 
  # ... 
}
```

And you could get named transactions for that endpoint. What do you think of that?

**Comment by akshah123** [comment]
Well, I had assumed that we could parse the request while before we had to name a given transaction.  I am mostly concerned with someone asking for multiple queries in a single request.  

Is it not possible to have the transaction name happen after parsing of graphql request is already done? 

In any case, I like the idea of using a flag. Though it would be better if it was done while enabling new relic instrumentation. Otherwise your example in controller works.

**Comment by rmosolgo** [comment]
That sounds good, we could also have 

```ruby 
  use GraphQL::Tracing::NewRelicTracing, tag_transactions: true 
```

I'll take a try and follow up here!

**Comment by rmosolgo** [comment]
Fixed by #1430
